src/etl/ETLSource.cpp

```diff
@@ -1159,7 +1159,7 @@ ETLLoadBalancer::execute(Func f, uint32_t ledgerSequence)
 
         log_.debug() << "Attempting to execute func. ledger sequence = "
                      << ledgerSequence << " - source = " << source->toString();
-        if (source->hasLedger(ledgerSequence) || true)
+        if (source->hasLedger(ledgerSequence))
```

Comment:
I am happy to see this line got remove !!!
but here is something we need to figure out first.
`
ETLLoadBalancer::loadInitialLedger(uint32_t sequence, bool cacheOnly)
{
    execute(
        [this, &sequence, cacheOnly](auto& source) {
            bool res =
                source->loadInitialLedger(sequence, downloadRanges_, cacheOnly);
            if (!res)
            {
                log_.error() << "Failed to download initial ledger."
                             << " Sequence = " << sequence
                             << " source = " << source->toString();
            }
            return res;
        },
        sequence);
}
`
The execute func is used to init the ledger. So "sequence" from "**loadIntialLedger**" is not in the range definitely. It might be root cause why we have this poison here. 
To fix this line, we need to refactor the larger scope of code.

Reply:
This line is just here because sometimes rippled has the ledger but doesn't actually know about it. This doesn't happen in the normal case and should be safe to remove. I believe I only needed this when loading full history in standalone mode.

Reply:
@officialfrancismendoza  To confirm it is okay to call loadInitialLedger in execute function with the actual check enable, can we have a test on emtpy db ?

Reply:
Thanks @cjcobb23  for clarification. 
@officialfrancismendoza  Can we also add CJ's comment on the code to note there is the rare case we might need it.

Reply:
Added relevant note and rebased @cindyyan317

Reply:
I'm slightly worried to remove this just based on CJ's knowledge and without unit-tests to support any assumptions but i'm willing to take the risk. Hopefully CJ is right and we will not hit this case in normal operation.